core/src/main/java/org/pac4j/jax/rs/pac4j/JaxRsContext.java

```diff
@@ -44,20 +49,24 @@
 
     private final ContainerRequestContext requestContext;
 
-    private final SessionStore sessionStore;
-
     private final Providers providers;
 
     private ResponseBuilder abortResponse = null;
 
     private MultivaluedMap<String, String> parameters = null;
 
+    private SessionStore sessionStore = null;
+
     public JaxRsContext(Providers providers, ContainerRequestContext requestContext, SessionStore sessionStore) {
```

Comment:
The `sessionStore` should not be used to create the `JaxRsContext`. Both should be treated separately.

Reply:
This is exactly how I see it. The reason why this was initially implemented this way (not by me) is that the Context is the only implementation-specific object that is available within the filters of the core project. I have already modified this so that the SessionStore is not needed in the constructor, but is determined by the context and the config from the providers. The only other alternatives would be to make the filters implementation-specific, to register the SessionStores as providers or to rely on the SessionStores from the Config. However, this would possibly break the compatibility.

Reply:
I don't know this implementation enough, but for sure, we can break the compatibility as it's a major version.
So if you have a better solution to properly separate the session store from the context, go ahead.